algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -338,6 +410,17 @@ case class DenseHLL(bits: Int, v: IndexedSeq[Byte]) extends HLL {
       idx += 1
     }
   }
+
+  override def downsize(reducedBits: Int): HLL = {
+    require(reducedBits > 3 && reducedBits < bits, s"Use at least 4, but less than $bits bits")
```

Comment:
again, let's go ahead and handle the `reducedBit == bits` case (like .take(0) is allowed, we might as well support this).